File: Sources/Teapot/Lexer/keywords.swift
import Foundation

func getKeywords() -> [String: Token] {
    return [
        "true": Token.ktrue,
        "false": Token.kfalse,
        "bool": Token.kbool,
        "int": Token.kint,
        "int32": Token.kint32,
        "int16": Token.kint16,
        "int8": Token.kint8,
        "uint": Token.kuint,
        "uint32": Token.kuint32,
        "uint16": Token.kuint16,
        "uint8": Token.kuint8,
        "float": Token.kfloat,
        "float32": Token.kfloat32
    ]
}

File: src/lexer/keywords.rs
use std::collections::HashMap;
use maplit::hashmap;
use super::types::TokenType;

pub fn get_keywords() -> HashMap<&'static str, TokenType> {
    return hashmap! {
        "true" => TokenType::True,
        "false" => TokenType::False,
        "print" => TokenType::Print,
        "bool" => TokenType::Bool,
        "int" => TokenType::Int,
        "int32" => TokenType::Int32,
        "int16" => TokenType::Int16,
        "int8" => TokenType::Int8,
        "uint" => TokenType::UInt,
        "uint32" => TokenType::UInt32,
        "uint16" => TokenType::UInt16,
        "uint8" => TokenType::UInt8,
        "float" => TokenType::Float,
        "float32" => TokenType::Float32
    };
}